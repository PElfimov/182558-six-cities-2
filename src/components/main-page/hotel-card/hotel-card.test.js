import React from 'react';
import HotelCard from './hotel-card';
import {BrowserRouter as Router} from 'react-router-dom';
import Enzyme, {shallow} from 'enzyme';
import toJSON from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({adapter: new Adapter()});

jest.mock(`../../../mocks/test-mocks`);

const offer = {
  isPremium: true,
  cost: 120,
  name: `Beautiful & luxurious apartment at great location`,
  rating: 55,
  type: `Apartment`
};

it(`HotelCard correctly renders after relaunch`, () => {
  const tree = shallow(<Router>
    <HotelCard
      offer={offer}
      onHover={jest.fn()} />
  </Router>);

  expect(toJSON(tree)).toMatchSnapshot();
});
