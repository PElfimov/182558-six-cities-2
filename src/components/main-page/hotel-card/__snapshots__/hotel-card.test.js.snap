// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HotelCard correctly renders after relaunch 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <HotelCard
    offer={
      Object {
        "cost": 120,
        "isPremium": true,
        "name": "Beautiful & luxurious apartment at great location",
        "rating": 55,
        "type": "Apartment",
      }
    }
    onHover={[MockFunction]}
  />
</Router>
`;
